libafl_frida/src/asan/asan_rt.rs

```diff
@@ -600,8 +569,9 @@ impl AsanRuntime {
                         let mut invocation = Interceptor::current_invocation();
                         let this = &mut *(invocation.replacement_data().unwrap().0 as *mut AsanRuntime);
                         let original = [<$name:snake:upper _PTR>].get().unwrap();
-
-                        if !ASAN_IN_HOOK.get() && this.hooks_enabled && this.[<hook_check_ $name>]($($param),*){
+                        //don't check if hooks are enabled as there are certain cases where we want to run the hook even if we are out of the program
```

Comment:
Can you elaborate? I don't quite understand.

Reply:
The case that I was facing is that very rarely there would be a case where while hooks were enabled and the fuzzer code was running, the fuzzer would make an allocation. This allocation would be made in our allocator. However, this allocation would be freed when hooks were disabled. Thus, it would try to free our own allocation in the normal allocator which an issue.

Reply:
Ok. We will have to check this behavior on windows...